File: Java/app/src/main/java/kz/sgq/fs_tiktaktoe/ui/activity/FieldActivity.java
package kz.sgq.fs_tiktaktoe.ui.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.arellomobile.mvp.MvpActivity;
import com.arellomobile.mvp.presenter.InjectPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;

import java.util.List;

import butterknife.BindView;
import butterknife.BindViews;
import butterknife.ButterKnife;
import butterknife.OnClick;
import kz.sgq.fs_tiktaktoe.R;
import kz.sgq.fs_tiktaktoe.mvp.presenter.FieldPresenter;
import kz.sgq.fs_tiktaktoe.mvp.view.FieldView;

public class FieldActivity extends MvpActivity implements FieldView {

    @InjectPresenter
    FieldPresenter presenter;

    @ProvidePresenter
    FieldPresenter provideFieldPresenter(){
        if (getIntent().getBooleanExtra("mode", true))
            return new FieldPresenter(true, getResources().getString(R.string.step),
                    getIntent().getStringExtra("playerOne"),
                    getIntent().getStringExtra("playerTwo"));
        else
            return new FieldPresenter(false, getResources().getString(R.string.step),
                    getIntent().getStringExtra("playerOne"),
                    getIntent().getStringExtra("playerTwo"));
    }

    @BindView(R.id.playerOne)
    TextView playerOne;

    @BindView(R.id.playerTwo)
    TextView playerTwo;

    @BindViews({R.id.box11, R.id.box12, R.id.box13,
            R.id.box21, R.id.box22, R.id.box23,
            R.id.box31, R.id.box32, R.id.box33})
    List<TextView> listBox;

    @BindView(R.id.step)
    TextView step;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_field);
        ButterKnife.bind(this);
    }

    @OnClick({R.id.box11, R.id.box12, R.id.box13,
            R.id.box21, R.id.box22, R.id.box23,
            R.id.box31, R.id.box32, R.id.box33,
            R.id.back, R.id.clear})
    public void onClickBox(View view) {
        switch (view.getId()) {
            case R.id.box11:
                presenter.onClickBox(0);
                break;
            case R.id.box12:
                presenter.onClickBox(1);
                break;
            case R.id.box13:
                presenter.onClickBox(2);
                break;
            case R.id.box21:
                presenter.onClickBox(3);
                break;
            case R.id.box22:
                presenter.onClickBox(4);
                break;
            case R.id.box23:
                presenter.onClickBox(5);
                break;
            case R.id.box31:
                presenter.onClickBox(6);
                break;
            case R.id.box32:
                presenter.onClickBox(7);
                break;
            case R.id.box33:
                presenter.onClickBox(8);
                break;
            case R.id.back:
                presenter.onClickBack();
                break;
            case R.id.clear:
                presenter.onClickClear();
                break;
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        presenter.onDestroy();
    }

    @Override
    public void setStep(String text) {
        step.setText(text);
    }

    @Override
    public void setVictoryOne(String victory) {
        playerOne.setText(victory);
    }

    @Override
    public void setVictoryTwo(String victory) {
        playerTwo.setText(victory);
    }

    @Override
    public void clearField() {
        for (int i = 0; i < listBox.size(); i++) {
            listBox.get(i).setText(null);
        }
    }

    @Override
    public void setListBox(int i, String text) {
        listBox.get(i).setText(text);
    }

    @Override
    public void finishActivity() {
        finish();
    }

    @Override
    public void textColor() {
        for (int i = 0; i < listBox.size(); i++) {
            listBox.get(i).setTextColor(getResources().
                    getColor(R.color.colorText));
        }
    }

    @Override
    public void textColorEnd(List<Integer> list) {
        for (int i = 0; i < listBox.size(); i++) {
            if (list.get(0) != i && list.get(1) != i &&
                    list.get(2) != i)
                listBox.get(i).setTextColor(getResources().
                        getColor(R.color.colorAccent));
        }
    }
}

File: Kotlin/app/src/main/java/kz/sgq/kotlintictactoe/ui/activity/FieldActivity.kt
package kz.sgq.kotlintictactoe.ui.activity

import android.os.Bundle
import android.view.View
import com.arellomobile.mvp.MvpActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import kotlinx.android.synthetic.main.activity_field.*
import kz.sgq.kotlintictactoe.R
import kz.sgq.kotlintictactoe.mvp.presenter.FieldPresenter
import kz.sgq.kotlintictactoe.mvp.view.FieldView

class FieldActivity : MvpActivity(), FieldView {

    @InjectPresenter
    lateinit var presenter: FieldPresenter

    @ProvidePresenter
    fun provideFieldPresenter(): FieldPresenter {
        return if (intent.getBooleanExtra("mode", true))
            FieldPresenter(true, getString(R.string.step),
                    intent.getStringExtra("playerOne"),
                    intent.getStringExtra("playerTwo"))
        else
            FieldPresenter(false, getString(R.string.step),
                    intent.getStringExtra("playerOne"),
                    intent.getStringExtra("playerTwo"))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_field)
    }

    fun onClickBox(view: View) {
        when (view.id) {
            R.id.box11 -> presenter.onClickBox(0)
            R.id.box12 -> presenter.onClickBox(1)
            R.id.box13 -> presenter.onClickBox(2)
            R.id.box21 -> presenter.onClickBox(3)
            R.id.box22 -> presenter.onClickBox(4)
            R.id.box23 -> presenter.onClickBox(5)
            R.id.box31 -> presenter.onClickBox(6)
            R.id.box32 -> presenter.onClickBox(7)
            R.id.box33 -> presenter.onClickBox(8)
            R.id.back -> presenter.onClickBack()
            R.id.clear -> presenter.onClickClear()
        }
    }

    override fun setStep(text: String) {
        step.text = text
    }

    override fun setVictoryOne(victory: String) {
        playerOne.text = victory
    }

    override fun setVictoryTwo(victory: String) {
        playerTwo.text = victory
    }

    override fun clearField() {
        box11.text = null
        box12.text = null
        box13.text = null
        box21.text = null
        box22.text = null
        box23.text = null
        box31.text = null
        box32.text = null
        box33.text = null
    }

    override fun setListBox(i: Int, text: String) {
        when (i) {
            0 -> box11.text = text
            1 -> box12.text = text
            2 -> box13.text = text
            3 -> box21.text = text
            4 -> box22.text = text
            5 -> box23.text = text
            6 -> box31.text = text
            7 -> box32.text = text
            8 -> box33.text = text
        }
    }

    override fun finishActivity() {
        finish()
    }

    override fun textColor() {
        box11.setTextColor(resources.getColor(R.color.colorText))
        box12.setTextColor(resources.getColor(R.color.colorText))
        box13.setTextColor(resources.getColor(R.color.colorText))
        box21.setTextColor(resources.getColor(R.color.colorText))
        box22.setTextColor(resources.getColor(R.color.colorText))
        box23.setTextColor(resources.getColor(R.color.colorText))
        box31.setTextColor(resources.getColor(R.color.colorText))
        box32.setTextColor(resources.getColor(R.color.colorText))
        box33.setTextColor(resources.getColor(R.color.colorText))
    }

    override fun textColorEnd(list: MutableList<Int>) {
        for (i in 0..8)
            if (list[0] != i && list[1] != i &&
                    list[2] != i)
                boxColor(i)
    }

    private fun boxColor(i: Int) {
        when (i) {
            0 -> box11.setTextColor(resources
                    .getColor(R.color.colorAccent))
            1 -> box12.setTextColor(resources
                    .getColor(R.color.colorAccent))
            2 -> box13.setTextColor(resources
                    .getColor(R.color.colorAccent))
            3 -> box21.setTextColor(resources
                    .getColor(R.color.colorAccent))
            4 -> box22.setTextColor(resources
                    .getColor(R.color.colorAccent))
            5 -> box23.setTextColor(resources
                    .getColor(R.color.colorAccent))
            6 -> box31.setTextColor(resources
                    .getColor(R.color.colorAccent))
            7 -> box32.setTextColor(resources
                    .getColor(R.color.colorAccent))
            8 -> box33.setTextColor(resources
                    .getColor(R.color.colorAccent))
        }
    }

}
